test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsTests.cs

```diff
@@ -37,100 +80,149 @@ public void OtlpExporterOptions_DefaultsForHttpProtobuf()
+        Environment.SetEnvironmentVariable(protocolEnvVarKeyName, "http/protobuf");
 
-        var options = new OtlpExporterOptions();
+        var options = new OtlpExporterOptions((OtlpExporterOptionsConfigurationType)configurationType);
 
         Assert.Equal(new Uri("http://test:8888"), options.Endpoint);
         Assert.Equal("A=2,B=3", options.Headers);
         Assert.Equal(2000, options.TimeoutMilliseconds);
         Assert.Equal(OtlpExportProtocol.HttpProtobuf, options.Protocol);
+        Assert.Equal(appendSignalPathToEndpoint, options.AppendSignalPathToEndpoint);
     }
 
-    [Fact]
-    public void OtlpExporterOptions_UsingIConfiguration()
+    [Theory]
+    [MemberData(nameof(GetOtlpExporterOptionsTestCases))]
+    public void OtlpExporterOptions_UsingIConfiguration(
+        int configurationType,
+        string endpointEnvVarKeyName,
+        string headersEnvVarKeyName,
+        string timeoutEnvVarKeyName,
+        string protocolEnvVarKeyName,
+        bool appendSignalPathToEndpoint)
     {
         var values = new Dictionary<string, string>()
         {
-            [OtlpExporterSpecEnvVarKeyDefinitions.DefaultEndpointEnvVarName] = "http://test:8888",
-            [OtlpExporterSpecEnvVarKeyDefinitions.DefaultHeadersEnvVarName] = "A=2,B=3",
-            [OtlpExporterSpecEnvVarKeyDefinitions.DefaultTimeoutEnvVarName] = "2000",
-            [OtlpExporterSpecEnvVarKeyDefinitions.DefaultProtocolEnvVarName] = "http/protobuf",
+            [endpointEnvVarKeyName] = "http://test:8888",
+            [headersEnvVarKeyName] = "A=2,B=3",
+            [timeoutEnvVarKeyName] = "2000",
+            [protocolEnvVarKeyName] = "http/protobuf",
         };
 
         var configuration = new ConfigurationBuilder()
             .AddInMemoryCollection(values)
             .Build();
 
-        var options = new OtlpExporterOptions(configuration, new());
+        var options = new OtlpExporterOptions(configuration, (OtlpExporterOptionsConfigurationType)configurationType, new());
 
         Assert.Equal(new Uri("http://test:8888"), options.Endpoint);
         Assert.Equal("A=2,B=3", options.Headers);
         Assert.Equal(2000, options.TimeoutMilliseconds);
         Assert.Equal(OtlpExportProtocol.HttpProtobuf, options.Protocol);
+        Assert.Equal(appendSignalPathToEndpoint, options.AppendSignalPathToEndpoint);
     }
 
     [Fact]
     public void OtlpExporterOptions_InvalidEnvironmentVariableOverride()
     {
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultEndpointEnvVarName, "invalid");
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultTimeoutEnvVarName, "invalid");
-        Environment.SetEnvironmentVariable(OtlpExporterSpecEnvVarKeyDefinitions.DefaultProtocolEnvVarName, "invalid");
+        var values = new Dictionary<string, string>()
+        {
+            ["InvalidEndpoint"] = "invalid",
```

Comment:
Why are we testing the behavior for invalid keys? This test was meant to test the behavior with invalid values, right? We would never provide invalid key names as parameters for `ApplyConfigurationUsingSpecificationEnvVars` anyway.

Reply:
Sorry those names were kind of misleading. There is nothing invalid about the names, only the values. I just tweaked the names so it should be less confusing now.